lib/libc/newlib/libc-hooks.c

```diff
@@ -303,10 +305,8 @@ void *_sbrk(intptr_t count)
 }
 __weak FUNC_ALIAS(_sbrk, sbrk, void *);
 
-#ifdef CONFIG_MULTITHREADING
```

Comment:
Uh, this isn't good -- what does this old version of newlib do for locking? Seems like this will cause all kinds of trouble if multiple threads call into stdio or anything else in newlib that requires locking.

Reply:
From what I have read, old newlib is not really thread safe. That's why they introduced the retargetable locks.